test/integration/test_cli_runners.py

```diff
@@ -0,0 +1,154 @@
+    key_file = tempfile.NamedTemporaryFile(suffix='_slurm_integration_ssh_key', delete=True).name
+    generate_key(key_file)
+    # Wait until ssh server in container is up
+    time.sleep(5)
+    with open("%s.pub" % key_file) as fh:
+        key = fh.read()
+    deploy_key(key=key, server='localhost', username='testuser', password='testuser', port=port)
+    return collections.namedtuple('remote_connection', 'hostname username password port private_key')(
+        'localhost', 'testuser', 'testuser', port, key_file
+    )
+
+
+def stop_ssh_docker(container_name):
+    subprocess.check_call(['docker', 'rm', '-f', container_name])
+
+
+def cli_job_config(remote_connection, shell_plugin='ParamikoShell', job_plugin='Slurm'):
+    job_conf_template = string.Template("""<job_conf>
+    <plugins>
+        <plugin id="cli" type="runner" load="galaxy.jobs.runners.cli:ShellJobRunner" workers="1"/>
+    </plugins>
+    <destinations default="ssh_slurm">
+        <destination id="ssh_slurm" runner="cli">
+            <param id="shell_plugin">$shell_plugin</param>
+            <param id="job_plugin">$job_plugin</param>
+            <param id="shell_username">$username</param>
+            <param id="shell_private_key">$private_key</param>
+            <param id="shell_hostname">$hostname</param>
+            <param id="shell_port">$port</param>
+            <param id="embed_metadata_in_job">False</param>
+            <env id="SOME_ENV_VAR">42</env>
+        </destination>
+    </destinations>
+</job_conf>
+""")
+    job_conf_str = job_conf_template.substitute(shell_plugin=shell_plugin,
+                                                job_plugin=job_plugin,
+                                                **remote_connection._asdict())
+    with tempfile.NamedTemporaryFile(suffix="_slurm_integration_job_conf", delete=False) as job_conf:
+        job_conf.write(job_conf_str)
+    return job_conf.name
+
+
+class BaseCliIntegrationTestCase(BaseJobEnvironmentIntegrationTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        if cls is BaseCliIntegrationTestCase:
+            raise unittest.SkipTest("Base class")
+        cls.container_name = "%s_container" % cls.__name__
+        cls.jobs_directory = tempfile.mkdtemp()
+        cls.remote_connection = start_ssh_docker(container_name=cls.container_name,
+                                                 jobs_directory=cls.jobs_directory)
+        super(BaseCliIntegrationTestCase, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        stop_ssh_docker(container_name=cls.container_name)
+        super(BaseCliIntegrationTestCase, cls).tearDownClass()
+
+    @classmethod
+    def handle_galaxy_config_kwds(cls, config, ):
+        config["jobs_directory"] = cls.jobs_directory
+        config["file_path"] = cls.jobs_directory
+        config["job_config_file"] = cli_job_config(remote_connection=cls.remote_connection, shell_plugin=cls.shell_plugin)
```

Comment:
Missing ` , job_plugin=cls.job_plugin` at the end of this line?

Reply:
oh yeah